src/fields2cover/types/Path.cpp

```diff
@@ -114,6 +114,81 @@ std::string Path::serializePath(size_t digit_precision) const {
   return res;
 }
 
+/**
+ * @brief Discretize the swath sections of the path and return a new path
+ * @param step_size Discretization step in [m]
+ * @param swath_speed Robot swath speed in [m/s]
+ * @return New path with swath now discretized 
+*/
+Path Path::discretize_swath(double step_size, double swath_speed) const {
+    // Create new path
+    Path new_path;
+
+    // Loop through all the points in the path
+    for (size_t i = 0; i < this->size(); i++)
+    {
+        // Check if the current point is a SWATH point
+        if (this->states.at(i).type == f2c::types::PathSectionType::SWATH)
+        {
+            // We know that swaths are represented by only two point so the next point will
+            // be the end of the swath
+            // Save start and end swath points to variables
+            Point start_point = this->states.at(i).point;
+            Point end_point = this->states.at(i+1).point;
+
+            // We can now discretize the swath
+            // Calculate the distance between the start and end point
+            double distance = start_point.Distance(end_point);
+            // Calculate the number of points needed to discretize the swath
+            int num_points = floor(distance / step_size);
```

Comment:
with floor, distances between points will be bigger than step_size. Why don't you use ceil?

Reply:
# Just a simple example to make sure we are on the same page:
- step_size = 3[m]
- swath_distance = 10[m]
- swath_start = 0[m]
- swath_end = 10[m]
## Floor
- num_points = floor(10/3) = floor(3.33) = 3 -> This will mean the new swath elements [0, 3, 6, 9, 10] thus the last step size will be the residual which is 1[m]
## Ceil
- num_points = ceil(10/3) = floor(3.33) = 4 -> This will mean the new swath elements [0, 3, 6, 9, 12] thus the last element is now 12[m] instead of 10[m]
## Other solution
Instead of using exactly the step size provided the closest approximate step size can be calculated to give equal steps between swaths with something like this:
- rounded_number_of_steps = std::round(swath_distance / step_size);

Case where step_size > swath_distance 

               // If rounded number of steps is zero, then the provided step_size is greater than the distance.
               // In this case, we should use the distance itself as the step size, thus no discretization needed.
               if(rounded_number_of_steps == 0) {
                     approx_step_size = distance;
                }

- approx_step_size = distance / rounded_number_of_steps

This will result in equal spaced steps from start to finish, but with a step_size that tries to approximate the user step_size. Let me know what you think.

Reply:
I went with the `Other solution`